tests/e2e-integration/shared/datasets/DatasetHelper.ts

```diff
@@ -0,0 +1,151 @@
+  }
+
+  static deaccession(persistentId: string) {
+    return cy
+      .visit(`/dataset.xhtml?persistentId=${persistentId}`)
+      .get('#editDataSet')
+      .click()
+      .get('#datasetForm\\:deaccessionDatasetLink')
+      .click()
+      .get('#datasetForm\\:reasonOptions_label')
+      .click()
+      .get('#datasetForm\\:reasonOptions_2')
+      .click()
+      .get('#datasetForm\\:reasonForDeaccession')
+      .type('Test deaccession')
+      .get('#datasetForm\\:j_idt2181')
+      .click()
+      .get('#datasetForm\\:deaccessionConfirmation_content > div > input')
+      .click()
+  }
+
+  static async createPrivateUrl(id: string): Promise<{ token: string }> {
+    return this.request<{ token: string }>(`/datasets/${id}/privateUrl`, 'POST')
+  }
+
+  static async createPrivateUrlAnonymized(id: string): Promise<{ token: string }> {
+    return this.request<{ token: string }>(
+      `/datasets/${id}/privateUrl?anonymizedAccess=true`,
+      'POST'
+    )
+  }
+
+  static async createWithFiles(
+    numberOfFiles: number,
+    useTabularFiles = false,
+    filesMetadata?: { [key: string]: string }
+  ): Promise<DatasetResponse> {
+    const datasetResponse = await this.create()
+    const files = await this.uploadFiles(
+      datasetResponse.persistentId,
+      numberOfFiles,
+      useTabularFiles,
+      filesMetadata
+    )
+    return { ...datasetResponse, files: files }
+  }
+
+  static async createWithFilesRestricted(
+    numberOfFiles: number,
+    useTabularFiles = false
+  ): Promise<DatasetResponse> {
+    const datasetResponse = await this.createWithFiles(numberOfFiles, useTabularFiles, {
+      description: 'This is an example file',
+      restrict: 'true'
+    })
+    return datasetResponse
+  }
+
+  static async createWithFilesEmbargoed(
+    numberOfFiles: number,
+    useTabularFiles = false
+  ): Promise<DatasetResponse> {
+    const datasetResponse = await this.createWithFiles(numberOfFiles, useTabularFiles, {
+      description: 'This is an example file',
+      restrict: 'true',
+      embargoDate: '2021-01-01'
+    })
+    return datasetResponse
+  }
+
+  static async embargoFiles(
+    persistentId: string,
+    filesIds: number[],
+    embargoDate: string
+  ): Promise<DatasetResponse> {
+    const response = await this.request<DatasetResponse>(
+      `/datasets/:persistentId/files/actions/:set-embargo?persistentId=${persistentId}`,
+      'POST',
+      { fileIds: filesIds, dateAvailable: embargoDate, reason: 'Standard project embargo' }
+    )
+    return { ...response, persistentId }
+  }
+
+  private static async uploadFiles(
+    datasetPersistentId: string,
+    numberOfFiles: number,
+    useTabularFile: boolean,
+    filesMetadata: { [key: string]: string } = { description: 'This is an example file' }
+  ): Promise<DatasetFileResponse[]> {
+    // TODO - Instead of uploading the files one by one, upload them all at once - do this refactor when integrating the pagination
```

Comment:
Isn't pagination already integrated? The only thing pending is the total number of files, but the pages work.

Reply:
I meant pagination working in the UI, and the pagination cannot work in the UI without the files total count

In any case, in the files count PR I couldn't find a way to upload several files at once via Native API
https://guides.dataverse.org/en/latest/api/native-api.html#add-a-file-to-a-dataset